pkg/boot/universalpayload/utilities_test.go

```diff
@@ -0,0 +1,285 @@
+			got, err := getFdtInfo(tt.name, tt.fdt)
+
+			if tt.err != nil {
+				if tt.err.Error() != err.Error() {
+					t.Errorf("getFdtInfo err = '%v', want '%v'", err, tt.err)
+				}
+				return
+			}
+
+			if err != nil {
+				t.Fatal(err)
+				return
+			}
+
+			if tt.fdtLoad != nil && got == nil {
+				t.Errorf("getFdtInfo fdtLoad = nil, want: %v", tt.fdtLoad)
+				return
+			}
+
+			if tt.fdtLoad != nil {
+				if tt.fdtLoad.Load != got.Load {
+					t.Errorf("getFdtInfo fdtLoad.Load = %d, want %v", got.Load, tt.fdtLoad.Load)
+				}
+				if tt.fdtLoad.EntryStart != got.EntryStart {
+					t.Errorf("getFdtInfo fdtLoad.EntryStart = %v, want %v", got.EntryStart, tt.fdtLoad.EntryStart)
+				}
+			}
+		})
+	}
+}
+
+func mockCPUTempInfoFile(t *testing.T, content string) string {
+	tmpDir := t.TempDir()
+	tempFile, err := os.CreateTemp(tmpDir, "cpuinfo")
+	if err != nil {
+		t.Fatalf("Failed to create temp file: %v", err)
+	}
+
+	sysfsCPUInfoPath = tempFile.Name()
+
+	if _, err := tempFile.WriteString(content); err != nil {
+		t.Fatalf("Failed to write to temp file: %v", err)
+	}
+
+	tempFile.Close()
+	return tempFile.Name()
+}
+
+func TestGetPhysicalAddressSizes(t *testing.T) {
+	tests := []struct {
+		name           string
+		cpuInfoContent string
+		expectedBits   uint8
+		expectedErr    string
+	}{
+		{
+			name: "Valid Address Size",
+			cpuInfoContent: `
+processor	: 0
+vendor_id	: GenuineIntel
+cpu family	: 6
+model		: 142
+model name	: Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz
+stepping	: 10
+microcode	: 0xea
+cpu MHz		: 1992.000
+cache size	: 8192 KB
+physical id	: 0
+siblings	: 4
+core id		: 0
+cpu cores	: 2
+apicid		: 0
+initial apicid	: 0
+address sizes	: 39 bits physical, 48 bits virtual
+`,
+			expectedBits: 39,
+			expectedErr:  "",
+		},
+		{
+			name: "No Address Size Info",
+			cpuInfoContent: `
+processor	: 0
+vendor_id	: GenuineIntel
+cpu family	: 6
+model		: 142value out of range
+model name	: Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz
+`,
+			expectedBits: 0,
+			expectedErr:  "address sizes information not found",
+		},
+		{
+			name: "Invalid Address Size",
+			// number value out of range
+			cpuInfoContent: `
+address sizes	: 1000 bits physical, 48 bits virtual
+`,
+			expectedBits: 0,
+			expectedErr:  "phyAddrSize 1000 out of range for uint8",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tempFile := mockCPUTempInfoFile(t, tt.cpuInfoContent)
+			defer os.Remove(tempFile)
+
+			physicalBits, err := getPhysicalAddressSizes()
+
+			if tt.expectedErr == "" {
+				// success validation
+				if err != nil {
+					t.Fatalf("Unexpected error: %v", err)
+				}
+				if physicalBits != tt.expectedBits {
+					t.Errorf("Unexpected physical address size %d, want = %d", physicalBits, tt.expectedBits)
+				}
+			} else {
+				// fault validation
+				if err == nil {
+					t.Fatalf("Expected error %q, got nil", tt.expectedErr)
+				}
+				if err.Error()[:len(tt.expectedErr)] != tt.expectedErr {
```

Comment:
Code like this is a sign that the package needs to be written with wrapped error returns.